src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -1965,6 +1968,27 @@ private bool LowerBoundInterfaceInference(TypeSymbol source, NamedTypeSymbol tar
             return true;
         }
 
+        internal static ImmutableArray<NamedTypeSymbol> ModuloNullabilityDifferences(IEnumerable<NamedTypeSymbol> interfaces, VarianceKind variance)
+        {
+            var dictionary = new Dictionary<NamedTypeSymbol, NamedTypeSymbol>(SymbolEqualityComparer.IgnoreNullableModifiersForReferenceTypes);
+
+            foreach (var @interface in interfaces)
+            {
+                if (dictionary.TryGetValue(@interface, out var found))
+                {
+                    dictionary.Remove(@interface);
+                    var merged = (NamedTypeSymbol)found.MergeNullability(@interface, variance);
+                    dictionary.Add(merged, merged);
```

Comment:
>merged [](start = 35, length = 6)

Consider adding original key and using values as the result.

Reply:
Then instead of removing-then-adding, we could use an update in place  ```dictionary[@interface] = merged```, which could be faster.

---
In reply to: [321434173](https://github.com/dotnet/roslyn/pull/38460#discussion_r321434173) [](ancestors = 321434173)